DataFormats/Detectors/EMCAL/include/DataFormatsEMCAL/AnalysisCluster.h

```diff
@@ -0,0 +1,226 @@
+
+  void setM02(double m02) { mM02 = m02; }
+  double getM02() const { return mM02; }
+
+  void setNExMax(unsigned char nExMax) { mNExMax = nExMax; }
+  unsigned char getNExMax() const { return mNExMax; }
+
+  void setEmcCpvDistance(double dEmcCpv) { mEmcCpvDistance = dEmcCpv; }
+  double getEmcCpvDistance() const { return mEmcCpvDistance; }
+  void setTrackDistance(double dx, double dz)
+  {
+    mTrackDx = dx;
+    mTrackDz = dz;
+  }
+  double getTrackDx(void) const { return mTrackDx; }
+  double getTrackDz(void) const { return mTrackDz; }
+
+  void setDistanceToBadChannel(double dist) { mDistToBadChannel = dist; }
+  double getDistanceToBadChannel() const { return mDistToBadChannel; }
+
+  void setNCells(int n) { mNCells = n; }
+  int getNCells() const { return mNCells; }
+
+  ///
+  ///  Set the array of cell absId numbers.
+  void setCellsAbsId(gsl::span<unsigned short> array);
+  gsl::span<unsigned short> getCellsAbsId() { return mCellsAbsId; }
+
+  ///
+  ///  Set the array of cell amplitude fractions.
+  ///  Cell can be shared between 2 clusters, here the fraction of energy
+  ///  assigned to each cluster is stored. Only in unfolded clusters.
+  void setCellsAmplitudeFraction(gsl::span<Double32_t> array);
+  gsl::span<Double32_t> getCellsAmplitudeFraction() { return mCellsAmpFraction; }
+
+  int getCellAbsId(int i) const
+  {
+    if (i >= 0 && i < mNCells)
+      return mCellsAbsId[i];
+    else
+      throw CellOutOfRangeException(i);
+  }
+
+  double getCellAmplitudeFraction(int i) const
+  {
+    if (i >= 0 && i < mNCells)
+      return mCellsAmpFraction[i];
+    else
+      throw CellOutOfRangeException(i);
+  }
+
+  bool getIsExotic() const { return mIsExotic; }
+  void setIsExotic(bool b) { mIsExotic = b; }
+
+  void setClusterTime(float time)
+  {
+    mTime = time;
+  }
+
+  float getClusterTime() const
+  {
+    return mTime;
+  }
+
+  int getIndMaxInput() const { return mInputIndMax; }
+  void setIndMaxInput(const int ind) { mInputIndMax = ind; }
+
+  float getCoreEnergy() const { return mCoreEnergy; }
+  void setCoreEnergy(float energy) { mCoreEnergy = energy; }
+
+  ///
+  /// Returns TLorentzVector with momentum of the cluster. Only valid for clusters
+  /// identified as photons or pi0 (overlapped gamma) produced on the vertex
+  /// Vertex can be recovered with esd pointer doing:
+  TLorentzVector getMomentum(std::array<const double, 3> vertexPosition) const;
+
+ protected:
+  gsl::span<int> mLabels; ///< List of MC particles that generated the cluster, ordered in deposited energy.
+
+  int mNCells = 0; ///< Number of cells in cluster.
+
+  /// Array of cell absolute Id numbers.
+  gsl::span<unsigned short> mCellsAbsId; //[mNCells]
+
+  /// Array with cell amplitudes fraction. Only usable for unfolded clusters, where cell can be shared.
+  /// here we store what fraction of the cell energy is assigned to a given cluster.
+  gsl::span<Double32_t> mCellsAmpFraction; //[mNCells][0.,1.,16]
```

Comment:
I am afraid this is not going to work, a span just wraps around another array. Both spans wrap around  vectors which would be owned by the cluster factory. Please change this to std::vector<int> and std::vector<Double_t 32>. Also for the MC labels it will not work like this, because MC labels in o2 are no longer integers but o2::MCLabel. We will have to refer here to the original cells, therefore I would consider it as better to refer to the cell indices instead of the cell abs ID.